Build fails when updating typescript from 4.9.5 to 5.0.2

### Description

```
tsconfig.json:10:5 - error TS5101: Option 'importsNotUsedAsValues' is deprecated and will stop functioning in TypeScript 5.5. Specify compilerOption '"ignoreDeprecations": "5.0"' to silence this error.
  Use 'verbatimModuleSyntax' instead.

12     "importsNotUsedAsValues": "error",
```
Source: https://github.com/callstack/react-native-builder-bob/blob/main/packages/create-react-native-library/templates/common/tsconfig.json#L10

### Packages

- [X] create-react-native-library
- [ ] react-native-builder-bob

### Selected options

N/A

### Link to repro

https://github.com/dcangulo/react-native-pkce-challenge/pull/241

### Environment

N/A

--- a/lefthook.yml
+++ b/lefthook.yml
@@ -7,7 +7,7 @@ pre-commit:
       run: npx eslint {files}
     types:
       files: git diff --name-only @{push}
-      glob: "*.{js,ts, jsx, tsx}"
+      glob: "*.{js,ts,jsx,tsx}"
       run: npx tsc --noEmit
 commit-msg:
   parallel: true

--- a/package.json
+++ b/package.json
@@ -28,7 +28,7 @@
     "eslint": "^8.18.0",
     "eslint-config-satya164": "^3.1.11",
     "prettier": "^2.7.1",
-    "typescript": "^4.7.4"
+    "typescript": "^5.0.2"
   },
   "commitlint": {
     "extends": [

--- a/packages/create-react-native-library/src/index.ts
+++ b/packages/create-react-native-library/src/index.ts
@@ -8,8 +8,9 @@ import spawn from 'cross-spawn';
 import ora from 'ora';
 import validateNpmPackage from 'validate-npm-package-name';
 import githubUsername from 'github-username';
-import prompts, { PromptObject } from './utils/prompts';
+import prompts from './utils/prompts';
 import generateExampleApp from './utils/generateExampleApp';
+import type { PromptObject } from './utils/prompts';
 
 const FALLBACK_BOB_VERSION = '0.20.0';
 

--- a/packages/create-react-native-library/src/index.ts
+++ b/packages/create-react-native-library/src/index.ts
@@ -8,9 +8,8 @@ import spawn from 'cross-spawn';
 import ora from 'ora';
 import validateNpmPackage from 'validate-npm-package-name';
 import githubUsername from 'github-username';
-import prompts from './utils/prompts';
+import prompts, { type PromptObject } from './utils/prompts';
 import generateExampleApp from './utils/generateExampleApp';
-import type { PromptObject } from './utils/prompts';
 
 const FALLBACK_BOB_VERSION = '0.20.0';
 

--- a/packages/create-react-native-library/templates/common/$package.json
+++ b/packages/create-react-native-library/templates/common/$package.json
@@ -82,7 +82,7 @@
 <% if (turborepo) { -%>
     "turbo": "^1.6.3",
 <% } -%>
-    "typescript": "^4.5.2"
+    "typescript": "^5.0.2"
   },
   "resolutions": {
     "@types/react": "17.0.21"

--- a/packages/create-react-native-library/templates/common/tsconfig.json
+++ b/packages/create-react-native-library/templates/common/tsconfig.json
@@ -7,7 +7,6 @@
     "allowUnreachableCode": false,
     "allowUnusedLabels": false,
     "esModuleInterop": true,
-    "importsNotUsedAsValues": "error",
     "forceConsistentCasingInFileNames": true,
     "jsx": "react",
     "lib": ["esnext"],
@@ -23,6 +22,7 @@
     "resolveJsonModule": true,
     "skipLibCheck": true,
     "strict": true,
-    "target": "esnext"
+    "target": "esnext",
+    "verbatimModuleSyntax": true
   }
 }

--- a/packages/create-react-native-library/templates/native-view-legacy/src/index.tsx
+++ b/packages/create-react-native-library/templates/native-view-legacy/src/index.tsx
@@ -1,9 +1,5 @@
-import {
-  requireNativeComponent,
-  UIManager,
-  Platform,
-  ViewStyle,
-} from 'react-native';
+import { requireNativeComponent, UIManager, Platform } from 'react-native';
+import type { ViewStyle } from 'react-native';
 
 const LINKING_ERROR =
   `The package '<%- project.slug -%>' doesn't seem to be linked. Make sure: \n\n` +

--- a/packages/create-react-native-library/templates/native-view-legacy/src/index.tsx
+++ b/packages/create-react-native-library/templates/native-view-legacy/src/index.tsx
@@ -1,5 +1,9 @@
-import { requireNativeComponent, UIManager, Platform } from 'react-native';
-import type { ViewStyle } from 'react-native';
+import {
+  requireNativeComponent,
+  UIManager,
+  Platform,
+  type ViewStyle,
+} from 'react-native';
 
 const LINKING_ERROR =
   `The package '<%- project.slug -%>' doesn't seem to be linked. Make sure: \n\n` +

--- a/packages/react-native-builder-bob/src/index.ts
+++ b/packages/react-native-builder-bob/src/index.ts
@@ -5,14 +5,13 @@ import dedent from 'dedent';
 import yargs from 'yargs';
 import { cosmiconfigSync } from 'cosmiconfig';
 import isGitDirty from 'is-git-dirty';
-import prompts from './utils/prompts';
+import prompts, { type PromptObject } from './utils/prompts';
 import * as logger from './utils/logger';
 import buildAAR from './targets/aar';
 import buildCommonJS from './targets/commonjs';
 import buildModule from './targets/module';
 import buildTypescript from './targets/typescript';
 import type { Options } from './types';
-import type { PromptObject } from './utils/prompts';
 
 // eslint-disable-next-line import/no-commonjs
 const { name, version } = require('../package.json');

--- a/packages/react-native-builder-bob/src/index.ts
+++ b/packages/react-native-builder-bob/src/index.ts
@@ -5,13 +5,14 @@ import dedent from 'dedent';
 import yargs from 'yargs';
 import { cosmiconfigSync } from 'cosmiconfig';
 import isGitDirty from 'is-git-dirty';
-import prompts, { PromptObject } from './utils/prompts';
+import prompts from './utils/prompts';
 import * as logger from './utils/logger';
 import buildAAR from './targets/aar';
 import buildCommonJS from './targets/commonjs';
 import buildModule from './targets/module';
 import buildTypescript from './targets/typescript';
 import type { Options } from './types';
+import type { PromptObject } from './utils/prompts';
 
 // eslint-disable-next-line import/no-commonjs
 const { name, version } = require('../package.json');

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -12,7 +12,6 @@
     "allowUnreachableCode": false,
     "allowUnusedLabels": false,
     "esModuleInterop": true,
-    "importsNotUsedAsValues": "error",
     "forceConsistentCasingInFileNames": true,
     "jsx": "react",
     "lib": ["esnext", "dom"],
@@ -28,7 +27,8 @@
     "resolveJsonModule": true,
     "skipLibCheck": true,
     "strict": true,
-    "target": "esnext"
+    "target": "esnext",
+    "verbatimModuleSyntax": true
   },
   "exclude": ["lib", "packages/create-react-native-library/templates"]
 }

